modules/ansible/ansibleee.go

```diff
@@ -0,0 +1,321 @@
+	// RestartPolicy default: Never
+	RestartPolicy string `json:"restartPolicy,omitempty"`
+	// CmdLine is the command line passed to ansible-runner
+	CmdLine string `json:"cmdLine,omitempty"`
+	// ServiceAccountName allows to specify what ServiceAccountName do we want the ansible execution run with. Without specifying,
+	// it will run with default serviceaccount
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+	// Inventory is the primary inventory that the ansible playbook will use to launch the job.
+	// Further inventories may be provided as ExtraMount in the `/runner/inventory/` path.
+	Inventory string `json:"inventory,omitempty"`
+	// Args are the command plus the playbook executed by the image. If args is passed, Playbook is ignored.
+	Args []string `json:"args,omitempty"`
+	// NetworkAttachments is a list of NetworkAttachment resource names to expose the services to the given network
+	NetworkAttachments []string `json:"networkAttachments,omitempty"`
+	// PreserveJobs - do not delete jobs after they finished e.g. to check logs
+	// PreserveJobs default: true
+	PreserveJobs bool `json:"preserveJobs,omitempty"`
+	// BackoffLimit allows to define the maximum number of retried executions (defaults to 6).
+	BackoffLimit *int32 `json:"backoffLimit,omitempty"`
+	// UID is the userid that will be used to run the container.
+	UID int64 `json:"uid,omitempty"`
+	// ExtraMounts containing conf files, credentials and inventories
+	ExtraMounts []storage.VolMounts `json:"extraMounts,omitempty"`
+	// InitContainers allows the passing of an array of containers that will be executed before the ansibleee execution itself
+	InitContainers []corev1.Container `json:"initContainers,omitempty"`
+	// DNSConfig allows to specify custom dnsservers and search domains
+	DNSConfig *corev1.PodDNSConfig `json:"dnsConfig,omitempty"`
+	// Extra vars to be passed to ansible process during execution. This can be used to override default values in plays.
+	ExtraVars map[string]json.RawMessage `json:"extraVars,omitempty"`
+	// Labels - Kubernetes labels to apply to the job
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations - Kubernetes annotations to apply to the job
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// Env is a list containing the environment variables to pass to the pod
+	Env []corev1.EnvVar `json:"env,omitempty"`
+}
+
+// EEJobInterface defines the functions required to format AnsibleEE kubernetes jobs
+type EEJobInterface interface {
+	JobForOpenStackAnsibleEE(h *helper.Helper) (*batchv1.Job, error)
+	addEnvFrom(job *batchv1.Job)
+	addMounts(job *batchv1.Job)
+}
+
+// JobForOpenStackAnsibleEE returns a openstackansibleee Job object
+func (a *EEJob) JobForOpenStackAnsibleEE(h *helper.Helper) (*batchv1.Job, error) {
+	ls := labelsForOpenStackAnsibleEE(a.Name, a.Labels)
+
+	args := a.Args
+
+	playbook := a.Playbook
+	if len(args) == 0 {
```

Comment:
Does having `args` > 0 implies that `len(playbook)` (that sets the default `playbook.yaml`) shouldn't be evaluated?

Reply:
Yeah, correct. If args are passed then `playbook` is ignored:
https://github.com/openstack-k8s-operators/lib-common/pull/536/files/ef2c08297fa11cf6c5b9cf71ebf1a86b48eb1204#diff-e150238f8cf75bd058263dacb9a03d71a789d1be75a9d0a45b0fcb3ba37bd8a6R44-R45